CONTRIBUTING.md

```diff
@@ -80,10 +80,17 @@ For any pull request submitted, the maintainers of Snap require `small` tests th
 
 ### Pull Request Guidelines
 
-Pull requests can contain a single commit or multiple commits. The most important part is that _**a single commit maps to a single fix**_. Here are a few scenarios:
-*  If a pull request adds a feature but also fixes two bugs, then the pull request should have three commits, one commit each for the feature and two bug fixes
-* If a PR is opened with 5 commits that was work involved to fix a single issue, it should be rebased to a single commit
-* If a PR is opened with 5 commits, with the first three to fix one issue and the second two to fix a separate issue, then it should be rebased to two commits, one for each issue
+The goal of following guidelines is to have Pull Requests (PRs) that are fairly easy to review and comprehend, and code that is easy to maintain in the future. Ideally those guidelines should be applied in any Snap related repositories (including plugins, plugin libraries, etc.).
+
+* **One PR addresses one problem** (not necessarily one whole GitHub Issue, for example in cases when single Issue describes several problems)
+* **Ideally one PR is associated with one commit** as it's trying to tackle one problem. Please squash your work to single commit before merge
+* **The PR should be small enough to be comprehended by human reviewer**. It's preferred to deliver subset of functionality (code) with tests and documentation rather than delivering code in one PR, then tests and documentation in subsequent PRs
+* **Ideally any code change shall be traceable to the GitHub Issue**, hence reference Issue in PR description. Any additional functionality not requested in GitHub Issue but delivered in the PR shall be explained in PR description
+* If you believe that the problem may be solved in several different ways, **describe design of the solution as part of the Issue, before submitting PR**
+* To make code changes easier to understand and review **consider giving hints to reviewers about scenario in which code is used**, also pointing out the impact outside changed code - can be done as "_Hints to reviewers_" comments provided by the PR author
+* **When code needs refactoring, also when it's not impacting user facing interfaces or user experience, provide reasoning and benefits of doing so, in the form of GitHub Issue**
+* **Don't forget about comments in the code** to help reviewers understand motivation or logic when it's not obvious
+* **If you submit a PR for review for which work is still in progress please indicate what is already available and what is still to be done**
```

Comment:
Again, bolded bullets is not easy to read and breaks the visual pattern set above. Maybe: 

* **Partial work is welcome** and you can submit it with the PR title including `[WIP]`